### PR TITLE
use CreateOrUpdate

### PR DESCRIPTION
**Pull request by jonas27**

### CODE REVIEW

File: internal/controller/server_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			}
+			Expect(k8sClient.Create(ctx, server)).Should(Succeed())
+
+			serverLookupKey := types.NamespacedName{Name: name, Namespace: namespace}
+			createdServer := &rampupv1alpha1.Server{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, serverLookupKey, createdServer)
+				return err == nil
+			}).Should(BeTrue())
```
Comment (avorima) [inline internal/controller/server_controller_test.go:47]: Should(Succeed()) and just use the error from Get

File: internal/controller/server_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	)
+
+	Context("When creating Server", func() {
+		It("Should create server", func() {
+			By("")
+			ctx := context.Background()
+			selector := make(map[string]string)
+			selector["test-label"] = "test"
+			server := &rampupv1alpha1.Server{
+				TypeMeta: metav1.TypeMeta{
```
Comment (avorima) [inline internal/controller/server_controller_test.go:29]: You don't need TypeMeta

File: internal/controller/suite_test.go [hunk 3]
```diff
@@ -38,16 +42,19 @@ var (
...
-func TestControllers(t *testing.T) {
+func TestControllers(t *testing.T) { //nolint:paralleltest
 	RegisterFailHandler(Fail)
 
 	RunSpecs(t, "Controller Suite")
 }
 
 var _ = BeforeSuite(func() {
 	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+	ctx, cancel = context.WithCancel(context.TODO())
```
Comment (avorima) [inline internal/controller/suite_test.go:57]: Prefer Background over TODO

File: internal/controller/suite_test.go [hunk 4]
```diff
@@ -38,16 +42,19 @@ var (
 	cfg       *rest.Config
 	k8sClient client.Client
 	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
 )
 
-func TestControllers(t *testing.T) {
+func TestControllers(t *testing.T) { //nolint:paralleltest
```
Comment (avorima) [inline internal/controller/suite_test.go:49]: You don't need these nolints

File: internal/controller/server_controller_test.go
Comment (avorima) [inline internal/controller/server_controller_test.go:1]: I'm not quite sure what this test is supposed to verify. You're basically doing the reconcile function's job

**Review by avorima — CHANGES_REQUESTED** [review]
The reconcile loop looks good. Just needs an update to make it compatible with the gRPC app

File: internal/controller/suite_test.go (hunk 3 above)
Comment (jonas27) [reply internal/controller/suite_test.go:57]: I had the opposite in my head, but for tests it is indeed true.

File: internal/controller/server_controller_test.go (hunk 2 above)
Comment (jonas27) [reply internal/controller/server_controller_test.go]: removed

**Pull request merged**